src/js/utils/hooks.js

```diff
@@ -42,3 +42,41 @@ export const useDidUpdateEffect = (onUpdate, deps) => {
         firstRef.current = true;
     }, deps);
 };
+
+export function getUrlSearchParams(key) {
+    const params = new URLSearchParams(window.location.search);
+    return params.get(key) || "";
+}
+
+export function updateUrlSearchParams(newTerm, key) {
+    const search = new URLSearchParams(window.location.search);
+    search.set(key, newTerm);
+    const newUrl = `${window.location.pathname}?${search.toString()}`;
+    window.history.replaceState({}, "", newUrl);
+}
```

Comment:
The `newUrl` variable is not actually a URL. It is a URL path. You can avoid the naming issue by just doing.
```suggestion
export function updateUrlSearchParams(newTerm, key) {
    const search = new URLSearchParams(window.location.search);
    search.set(key, newTerm);
    window.history.replaceState({}, "", `${window.location.pathname}?${search.toString()}`);
}
```